lib/truncate.js

```diff
@@ -8,123 +9,141 @@ exports.span = truncSpan
 exports.error = truncError
 
 function truncMetadata (metadata, opts) {
-  metadata.process.title = truncate(String(metadata.process.title), opts.truncateKeywordsAt)
+  breadthFilter(metadata, (value, key, path) => {
+    if (typeof value !== 'string') {
+      return value
+    }
+
+    let max = opts.truncateStringsAt
+    if (path.join('.') === 'process.title') {
+      max = opts.truncateKeywordsAt
+    }
+
+    return truncate(value, max)
+  }, true)
 }
 
 function truncTransaction (trans, opts) {
   trans.name = truncate(String(trans.name), opts.truncateKeywordsAt)
   trans.type = truncate(String(trans.type), opts.truncateKeywordsAt)
   trans.result = truncate(String(trans.result), opts.truncateKeywordsAt)
 
-  // Unless sampled, context will be null
-  if (trans.sampled) truncContext(trans.context, opts.truncateKeywordsAt)
+  breadthFilter(trans, (value, key, path) => {
+    if (typeof value !== 'string') {
+      return value
+    }
+
+    let max = opts.truncateStringsAt
+    if (trans.sampled && path[0] === 'context') {
+      max = contextLength(path.slice(1), opts)
+    }
+    switch (path.join('.')) {
```

Comment:
I think this could just be replaced by `path[0]` right?

Reply:
fixed